### ISSUE TITLE
ot-br-posix built with Yocto cannot ping/reach external addresses.

### ISSUE THREAD

**Issue by ArtLiW** [openthread/ot-br-posix#1322]
Hi all,
we tried to build ot-br-posix (58d09bee8) based on Yocto recipe provided in: https://github.com/openthread/ot-br-posix/blob/main/etc/yocto/otbr_git.bb by @gabekassel.

We were able to build an OS image for the raspberry-pi with a SiLabs RCP and start the otbr-agent service and create a Thread network using CLI commands. We could also join one or more nodes to the Thread network and ping between them. 

When we used the manual installation we were able to ping e.g. google (ping 64:ff9b::808:808) with above mentioned hardware setup (raspi + SiLabs RCP) but with the Yocto recipe based version this is not possible.

By comparing the two different setups, manual (working google ping) and Yocto based (not working google ping) we observed that in the manual setup a nat64 interface is created. Not so for the Yocto based setup. Screenshot of the manual setup is attached. Parts marked in red are missing in the Yocto recipe based setup.

Another observation is that the tayga/NAT64 service in the manual setup is running. In the Yocto-based OTBR the tayga/NAT64 service is not running. However, we still haven't managed to create the nat64 interface that tayga needs to work.

It seems that something (tayga/NAT64 + interface?) is missing in our Yocto recipe which would enable external communication over the border router.

Can someone advise please?

Thanks a lot!
ArtLiW

**Expected behavior** A clear and concise description of what you expected to happen.
ot-br-posix built with Yocto able to communicate with external/internet addresses e.g. ping 64:ff9b::808:808

**Console/log output** If applicable, add console/log output to help explain your problem.
ping 64:ff9b::808:808
results in
"Packet loss = 100.0%"
![ifconfig_manual_](https://user-images.githubusercontent.com/102662258/160879761-b5183f49-bdab-4659-9c69-b619f1207a1e.jpg)

**Comment by jwhui** [comment]
@ArtLiW , not sure if it helps, but the setup script that configures NAT64 is [`script/_nat64`](https://github.com/openthread/ot-br-posix/blob/main/script/_nat64).

**Comment by gabekassel** [comment]
+1 to @jwhui 's recommendation. The yocto recipe provided in OTBR only helps install otbr-agent and ot-ctl. Tayga, avahi-daemon, and other helper services need to be installed and started separately. 

Further, you'll still need to add a NAT64 route to the thread network data (e.g. `ot-ctl route add FOO n`) and configure tayga appropriately. Happy to share examples of a tayga configuration file if helpful, but I think the suggested otbr setup scripts should help.

**Comment by ArtLiW** [comment]
Hi @jwhui and @gabekassel,

thanks a lot for your hints and help! 
We were able to resolve the external connectivity problem by adding two IP rules in the nat table:

iptables -t nat -A POSTROUTING -s 192.168.255.0/24 -j MASQUERADE

iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE

We added above two lines between the two following lines in our script and it worked:

ip route add 192.168.255.0/24 dev nat64

(add here)

systemctl start tayga.service

Regarding the rest of the configuration we have some concerns about our DNS64. We are trying to understand how we can test the DNS64 configuration and if it is possible to do “ping www.google.com" with that configuration from our end devices. The same applies to the NetworkManager and DHCP configurations. We don’t know how to test these configurations.
It would be great if that would be explained in the Openthread Border Router documentation.

Please let me know if this is already available.

Thanks again!!

Best Regards,

**Comment by jwhui** [comment]
Closing stale issue.
